File: services/app/dockers/ruby/checker.rb
require 'json'
require 'test/unit'
$stdout = STDERR

extend Test::Unit::Assertions

checks = []

STDIN.read.split("\n").each do |line|
  checks.push(JSON.parse(line))
end

begin
  require './check/solution'

  checks.each do |element|
    if element['check']
      puts JSON.dump(
        status: :ok,
        result: element['check']
      )
    else
      result = method(:solution).call(*element['arguments'])
      begin
        assert_equal(element['expected'], result)
      rescue Test::Unit::AssertionFailedError
        puts JSON.dump(
          status: :failure,
          result: element['arguments']
        )
        exit(0)
      end
    end
  end
rescue Exception => e
  puts(JSON.dump(
         status: :error,
         result: e.message
       ))
  exit(0)
end

File: services/app/dockers/python/checker.py
import sys
import json

checks = [json.loads(l) for l in sys.stdin.read().split("\n")]

try:
    from check.solution import solution
    for element in checks:
        if 'check' in element.keys():
            print(json.dumps({
                'status': 'ok',
                'result': element['check'],
            }))
        else:
            assert solution(*element['arguments']) == element['expected'], element['arguments']

except AssertionError as exc:
    print(json.dumps({
        'status': 'failure',
        'result': exc.args[0],
    }))
    exit(0)
except Exception as exc:
    print(json.dumps({
        'status': 'error',
        'result': 'unexpected',
    }))
    exit(0)
